client/containers/Pub/SpubHeader/SpubHeader.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import { Tab, Tabs, TabId, Icon, IconName } from '@blueprintjs/core';
+
+import InstructionsTab from './InstructionsTab';
+import SubmissionTab from './SubmissionTab';
+import PreviewTab from './PreviewTab';
+
+require('./spubHeader.scss');
+
+type Props = {
+	updateLocalData: any;
+	historyData: any;
+	pubData: any;
+};
+
+export const renderInstructionTabTitle = (icon: IconName, title: string) => {
+	return (
+		<>
+			<Icon icon={icon} /> {title}
+		</>
+	);
+};
+
+const SpubHeader = (props: Props) => {
+	const { submissionWorkflow } = props.pubData.submission || {};
+	const [selectedTab, setSelectedTab] = useState<TabId>('instructions');
+	if (!submissionWorkflow) return null;
```

Comment:
I'm not sure if this was your change or the file was just moved/renamed, so feel free to ignore this feedback!

Can we assert that the `submission` property is always there? I feel like the check here gives a false impression that this component is meant to be rendered with or without a submission (e.g. a submission-less Pub would render some non-ideal state, rather than `null`). Whereas I feel an error _should_ occur if we pass a submission-less pub.

I think we talked about this last week and decided on "it's a matter of preference". I tend to lean towards throwing well-defined errors rather than coding defensively because I find the code a bit easier to reason about.

Another way to handle this is to accept a `submissionWorkflow` prop instead of reading it off of `pubData`. Thoughts?

Reply:
I'm into keeping the `pubData` approach, in that it follows the pattern of the `PubHeader` more closely; keeping them as close evolutionary cousins seems helpful.

Yes, submission is always there. Holdover check from some more brittle state of the code, I think. But you're right, an error for a missing `submissionWorkflow` would be more instructive. I'm a little fuzzy on what kind of promises we can make about the data passed into `container`s overall -- seems like `pubData` and `pageData` get the `any` type often, and it makes me a little queasy after so I've become so dependent on typesafety everywhere else.